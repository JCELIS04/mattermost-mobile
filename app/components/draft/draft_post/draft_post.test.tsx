// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {renderWithEverything} from '@test/intl-test-helper';
import TestHelper from '@test/test_helper';

import DraftPost from '.';

import type {Database} from '@nozbe/watermelondb';
import type DraftModel from '@typings/database/models/servers/draft';

describe('Draft Post', () => {
    let database: Database;

    beforeAll(async () => {
        const server = await TestHelper.setupServerDatabase();
        database = server.database;
    });
    it('should match the snapshot', () => {
        const props = {
            draft: {
                updateAt: 1633024800000,
                message: 'Hello, World!',
                channelId: 'channel_id',
                rootId: '',
                files: [],
                metadata: {},
            } as unknown as DraftModel,
            layoutWidth: 100,
            location: 'draft',
        };
        const wrapper = renderWithEverything(
            <DraftPost {...props}/>, {database},
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
