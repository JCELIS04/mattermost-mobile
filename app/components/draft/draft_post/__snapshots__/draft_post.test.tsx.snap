// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draft Post should match the snapshot 1`] = `
<View
  style={
    {
      "marginTop": 12,
    }
  }
  testID="draft_post_with_message_and_file"
>
  <View
    style={
      {
        "maxHeight": 721,
      }
    }
    testID="draft_message"
  >
    <RCTScrollView
      keyboardShouldPersistTaps="always"
      scrollEnabled={false}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            [
              {
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                },
              ]
            }
            testID="markdown_paragraph"
          >
            <Text>
              <Text
                selectable={false}
                style={
                  {
                    "color": "#3f4350",
                    "fontFamily": "OpenSans",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  }
                }
                testID="markdown_text"
              >
                Hello, World!
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
